src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -303,15 +305,38 @@ public void loadData() throws IOException, InterruptedException {
 
     public void takeSnapshot(){
```

Comment:
+1. This is the reason why I advocated the use of `AtomicBoolean` instead of `AtomicInteger` to provide mutual exclusion on this code snippet.